Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js

```diff
@@ -131,6 +132,11 @@ WI.DebuggerManager = class DebuggerManager extends WI.Object
                 const key = null;
                 WI.objectStores.breakpoints.associateObject(breakpoint, key, serializedBreakpoint);
 
+                // HACK: Each breakpoint needs to have a linked sourceCode for it to be able to show up in the source sidebar, and the `breakpointsForSourceCode` method normally does the linking job. However, when the inspector opens and the list of breakpoints is first loaded, there is a race between that method runs and this code runs. If that function runs first, breakpoint loaded here won't get linked to its sourceCode, resulting it missing from the sidebar. This code gets around the race condition and does the sourceCode linking here too.
```

Comment:
We don't use `HACK` for comments, and if my question below leads to this being the only place we run this code then we should remove the comment entirely and say the relevant info in the commit message instead.

For comments please also manually line break at some reasonable character limit like 120 chars.

Also in general we should do this work once. Does it work to only do it here instead of `breakpointsForSourceCode`?

Reply:
honestly the whole idea of having to even do this manual association seems hacky (or at least not ideal).  i would've hoped/preferred that we just do everything based on the `contentIdentifier` instead

with that being said, i think what @patrickangle was maybe trying to get at (and please correct me if i'm wrong) was that this only runs once when Web Inspector is first opened, so it's not ideal to keep `_breakpointSourceCodeContentIdentifierMap` around beyond that point as it'll never be used again

perhaps we should have a `_breakpointsInitialized` flag that's set above which is used in `breakpointsForSourceCode` to prevent anything from being added to  `_breakpointSourceCodeContentIdentifierMap` (which can also be cleared right after this instead of in `_mainResourceDidChange`)?